### ISSUE TITLE
Warn about missing expand_aliases

### ISSUE THREAD

**Issue by piegamesde** [koalaman/shellcheck#2751]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related
  - There are a lot of alias-related bugs, but none about the general use of them

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

alias foo="bar"
foo
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

https://askubuntu.com/a/1057079/549930

Aliases are not expanded when the shell is not interactive, unless the expand_aliases shell option is set using shopt

Maybe also suggest to use a function instead of the alias
